launcher/electron/app/window-manager.js

```diff
@@ -0,0 +1,135 @@
+/*
+ *  Copyright 2015 TWO SIGMA OPEN SOURCE, LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+module.exports = (function() {
+  var BrowserWindow = require('browser-window');
+  var backendRunner = require('./backend-runner.js');
+  var ipc = require('ipc');
+  var Faye = require('faye');
+
+  var _windows = {};
+  var _windowToSession = {};
+  var _sessionToWindow = {};
```

Comment:
So _windows is just a collection of all windows handled by electron. It is only there because if a global reference is not kept, then windows get garbage collected.